lib/mutant/repository.rb

```diff
@@ -61,6 +63,26 @@ def touches?(path, line_range)
         !stdout.empty?
       end
 
+    private
+
+      # Test if path is tracked in repository
+      #
+      # FIXME: Cache results, to avoid spending time on producing redundant results.
+      #
+      # @param [Pathname] path
+      #
+      # @return [Boolean]
+      #
+      # @api private
+      def tracks?(path)
+        command = %W[git ls-files --error-unmatch -- #{path}]
+        Kernel.system(
+          *command,
+          out: File::NULL,
```

Comment:
It reduces the need to shell out, and build shell commands. Thats great as no escaping is required. (Still beware for option injection).